lnrpc/neutrinorpc/neutrino.proto

```diff
@@ -0,0 +1,214 @@
+syntax = "proto3";
+
+package neutrinorpc;
+
+option go_package = "github.com/lightningnetwork/lnd/lnrpc/neutrinorpc";
+
+// Neutrino is a service that can be used to get information about the current
+// state and scan for transactions.
+service NeutrinoKit {
+    /*
+    Status returns the status of the light client neutrino instance,
+    along with height and hash of the best block, and a list of connected
+    peers.
+    */
+    rpc Status (StatusRequest) returns (StatusResponse);
+
+    /*
+    AddPeer adds a new peer that has already been connected to the server.
+    */
+    rpc AddPeer (AddPeerRequest) returns (AddPeerResponse);
+
+    /*
+    DisconnectPeer disconnects a peer by target address. Both outbound and
+    inbound nodes will be searched for the target node. An error message will
+    be returned if the peer was not found.
+    */
+    rpc DisconnectPeer (DisconnectPeerRequest) returns (DisconnectPeerResponse);
+
+    /*
+    IsBanned returns true if the peer is banned, otherwise false.
+    */
+    rpc IsBanned (IsBannedRequest) returns (IsBannedResponse);
+
+    /*
+    GetBlockHeader returns a block header with a particular block hash.
+    */
+    rpc GetBlockHeader (GetBlockHeaderRequest) returns (GetBlockHeaderResponse);
+
+    /*
+    GetBlock returns a block with a particular block hash.
+    */
+    rpc GetBlock (GetBlockRequest) returns (GetBlockResponse);
+
+    /*
+    GetCFilter returns a compact filter from a block.
+    */
+    rpc GetCFilter (GetCFilterRequest) returns (GetCFilterResponse);
+}
+
+message StatusRequest {
+}
+
+message StatusResponse {
+    // Indicates whether the neutrino is active or not.
+    bool active = 1;
+
+    // Is fully synced.
+    bool synced = 2;
+
+    // Best block height.
+    int32 block_height = 3;
+
+    // Best block hash.
+    string block_hash = 4;
+
+    // Connected peers.
+    repeated string peers = 5;
+}
+
+message AddPeerRequest {
+    // Peer to add.
+    string peer_addrs = 1;
+}
+
+message AddPeerResponse {
+}
+
+message DisconnectPeerRequest {
+    // Peer to disconnect.
+    string peer_addrs = 1;
+}
+
+message DisconnectPeerResponse {
+}
+
+message IsBannedRequest {
+    // Peer to lookup.
+    string peer_addrs = 1;
+}
+
+message IsBannedResponse {
+    bool banned = 1;
+}
+
+message GetBlockHeaderRequest {
+    // Block hash in hex notation.
+    string hash = 1;
```

Comment:
Any reason to prefer this over accepting the raw bytes?

Reply:
I did try to mimic was bitcoind is returning in the getblockheader response (https://chainquery.com/bitcoin-cli/getblockheader).